tests/dependency_analyzer/cyclonedx/test_cyclonedx.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module tests the CyclondeDX helper functions."""
+
+import json
+import os
+from pathlib import Path
+
+from macaron.dependency_analyzer.cyclonedx import (
+    CycloneDXParserError,
+    convert_components_to_artifacts,
+    deserialize_bom_json,
+    get_dep_components,
+)
+from tests.macaron_testcase import MacaronTestCase
+
+
+class TestCyclondeDX(MacaronTestCase):
+    """Test the CyclondeDX helper functions."""
+
+    RESOURCES_DIR = Path(__file__).parent.joinpath("resources")
+    EXPECTED_DIR = Path(__file__).parent.joinpath("expected_results")
+
+    def test_deserialize_bom_json(self) -> None:
+        """Test deserializing a bom.json file."""
+        # Deserialize a valid bom.json.
+        path = Path(TestCyclondeDX.RESOURCES_DIR, "valid_bom.json")
+        deserialize_bom_json(path)
+
+        # Deserialize a bom.json that does not exist.
+        path = Path(TestCyclondeDX.RESOURCES_DIR, "does_not_exist")
+        self.assertRaises(CycloneDXParserError, deserialize_bom_json, path)
+
+        # Deserialize an invalid bom.json.
+        path = Path(TestCyclondeDX.RESOURCES_DIR, "invalid_bom.json")
+        self.assertRaises(CycloneDXParserError, deserialize_bom_json, path)
+
+    @classmethod
+    def test_get_dep_components(cls) -> None:
```

Comment:
Why would we need to make this a class method instead of using `self` to access the `RESOURCES_DIR` attribute?
For example:
```python
    def test_get_dep_components(self) -> None:
        self.RESOURCES_DIR ...
```

Reply:
`self.x` creates a new object and overrides the class variable `RESOURCES_DIR`. It's a common mistake that can be avoided by using `cls.RESOURCES_DIR`.

Anyway, I am going to refactor this whole test and use `pytest` and `syrupy` to simplify.